subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c

```diff
@@ -423,18 +427,29 @@ static void nct_reset_topics(void)
 
 static void nct_topic_lists_populate(void)
 {
-	/* Add RX topics */
-	nct_cc_rx_list[0].qos = MQTT_QOS_1_AT_LEAST_ONCE;
-	nct_cc_rx_list[0].topic.utf8 = accepted_topic;
-	nct_cc_rx_list[0].topic.size = strlen(accepted_topic);
-
-	nct_cc_rx_list[1].qos = MQTT_QOS_1_AT_LEAST_ONCE;
-	nct_cc_rx_list[1].topic.utf8 = rejected_topic;
-	nct_cc_rx_list[1].topic.size = strlen(rejected_topic);
-
-	nct_cc_rx_list[2].qos = MQTT_QOS_1_AT_LEAST_ONCE;
-	nct_cc_rx_list[2].topic.utf8 = update_delta_topic;
-	nct_cc_rx_list[2].topic.size = strlen(update_delta_topic);
+	/* Add RX topics, aligning with opcode list */
+	for (int idx = 0; idx < CC_RX_LIST_CNT; ++idx) {
+		if (nct_cc_rx_opcode_map[idx] == NCT_CC_OPCODE_UPDATE_ACCEPTED) {
+			nct_cc_rx_list[idx].qos = MQTT_QOS_1_AT_LEAST_ONCE;
+			nct_cc_rx_list[idx].topic.utf8 = accepted_topic;
+			nct_cc_rx_list[idx].topic.size = strlen(accepted_topic);
+			continue;
+		}
+
+		if (nct_cc_rx_opcode_map[idx] == NCT_CC_OPCODE_UPDATE_REJECTED) {
+			nct_cc_rx_list[idx].qos = MQTT_QOS_1_AT_LEAST_ONCE;
+			nct_cc_rx_list[idx].topic.utf8 = rejected_topic;
+			nct_cc_rx_list[idx].topic.size = strlen(rejected_topic);
+			continue;
+		}
+
+		if (nct_cc_rx_opcode_map[idx] == NCT_CC_OPCODE_UPDATE_DELTA) {
+			nct_cc_rx_list[idx].qos = MQTT_QOS_1_AT_LEAST_ONCE;
+			nct_cc_rx_list[idx].topic.utf8 = update_delta_topic;
+			nct_cc_rx_list[idx].topic.size = strlen(update_delta_topic);
+			continue;
+		}
```

Comment:
Should we add an assert here at the end in case `idx` passes through an un-handled opcode value? Main benefit being preventing someone from forgetting to implement a definition here. Admittedly, this does seem unlikely, so maybe not worth it